### PR TITLE
feat(ui): socket debug mode, network cleanup

### PR DESCRIPTION
**Pull request by psychedelicious**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

## Description

feat(ui): add debug mode & socketOptions

- Add `isDebugging` prop to `<InvokeAIUI />`. If `isDebugging` is `true`, you may configure the socket client from the JS console.
  - Ex `<InvokeAIUI isDebugging={true} />`
- Add `socketOptions` prop to `<InvokeAIUI />`. This provides the initial, base options.
  - Ex `<InvokeAIUI socketOptions={{ reconnectionDelay: 1000, reconnectionDelayMax: 5000 }} />`

### `$socketOptions`

This is a [nanostores `map`](https://github.com/nanostores/nanostores#maps).

The object may contain any of the [Socket.IO client options](https://socket.io/docs/v4/client-options/).

TL;DR `map` API:
- `$socketOptions.get()`: get the options object itself
- `$socketOptions.set()`: set the whole object - you *must* reference the original object, and overwrite individual properties
   - Ex `$socketOptions.get({...$socketOptions.get(), reconnectionDelay: 5000, reconnectionDelayMax: 10000})`
   - Note the spread operator `...$socketOptions.get()`
- `$socketOptions.setKey('optionName', value)`: set a specific option
   - Ex `$socketOptions.setKey('reconnectionDelay', 5000)`

### `$socketUrl`

This is a [nanostores `atom`](https://github.com/nanostores/nanostores#atoms).

The object may be assigned a string. If you assign an empty string, it will use the default.

TL;DR `atom` API:
- `$socketUrl.get()`: gets the current value
- `$socketUrl.set('new url')`: sets the value

### I broke it or I want to reset everything

Refresh the page

### Reactivity

When you change an option or url, the socket will teardown and re-initialize itself. You'll get a log with the `socket` client for your inspection pleasure.

### `$socketOptions` doesn't exist

Make sure you've passed `isDebugging={true}` to the UI.

### CODE REVIEW

**Comment by psychedelicious** [comment]
This PR also moves the socket.io logic into a hook, instead of the convoluted middleware.

**Comment by psychedelicious** [comment]
I've organised some network related files and updated the frontend readme.

**Review by hipsterusername — APPROVED** [review]
